### ISSUE TITLE
[BUG] om.azure.core.exception.HttpResponseException: Deserialization Failed.

### ISSUE THREAD

**Issue by shweta5595** [Azure/azure-sdk-for-java#41869]
**Describe the bug**
I am using azure-storage-blob v12.16.0 dependency to get storage blobs data. It was working fine until last month, but now I am getting following exception:
com.azure.core.exception.HttpResponseException: Deserialization Failed.
***Exception or Stack Trace***
```
com.azure.core.exception.HttpResponseException: Deserialization Failed.
	at com.azure.core.implementation.serializer.HttpResponseBodyDecoder.decodeByteArray(HttpResponseBodyDecoder.java:99) ~[azure-core-1.49.0.jar:1.49.0]
	at com.azure.core.implementation.serializer.HttpResponseDecoder$HttpDecodedResponse.getDecodedBody(HttpResponseDecoder.java:93) ~[azure-core-1.49.0.jar:1.49.0]
	at com.azure.core.implementation.http.rest.AsyncRestProxy.lambda$decodeBytes$8(AsyncRestProxy.java:169) ~[azure-core-1.49.0.jar:1.49.0]
	at reactor.core.publisher.MonoCallable.call(MonoCallable.java:92) ~[reactor-core-3.4.29.jar:3.4.29]
	at reactor.core.publisher.FluxSubscribeOnCallable$CallableSubscribeOnSubscription.run(FluxSubscribeOnCallable.java:227) ~[reactor-core-3.4.29.jar:3.4.29]
	at reactor.core.scheduler.SchedulerTask.call(SchedulerTask.java:68) ~[reactor-core-3.4.29.jar:3.4.29]
	at reactor.core.scheduler.SchedulerTask.call(SchedulerTask.java:28) ~[reactor-core-3.4.29.jar:3.4.29]
	at java.base/java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:264) ~[na:na]
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java) ~[na:na]
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[na:na]
	at java.base/java.lang.Thread.run(Thread.java:829) ~[na:na]
	Suppressed: java.lang.Exception: #block terminated with an error
		at reactor.core.publisher.BlockingSingleSubscriber.blockingGet(BlockingSingleSubscriber.java:99) ~[reactor-core-3.4.29.jar:3.4.29]
		at reactor.core.publisher.Flux.blockLast(Flux.java:2761) ~[reactor-core-3.4.29.jar:3.4.29]
		at com.azure.core.util.paging.ContinuablePagedByIteratorBase.requestPage(ContinuablePagedByIteratorBase.java:102) ~[azure-core-1.49.0.jar:1.49.0]
		at com.azure.core.util.paging.ContinuablePagedByItemIterable$ContinuablePagedByItemIterator.<init>(ContinuablePagedByItemIterable.java:75) ~[azure-core-1.49.0.jar:1.49.0]
		at com.azure.core.util.paging.ContinuablePagedByItemIterable.iterator(ContinuablePagedByItemIterable.java:55) ~[azure-core-1.49.0.jar:1.49.0]
		at java.base/java.lang.Iterable.spliterator(Iterable.java:101) ~[na:na]
		at com.azure.core.util.paging.ContinuablePagedIterable.stream(ContinuablePagedIterable.java:85) ~[azure-core-1.49.0.jar:1.49.0]
		at com.sams.cash.DLQProcessor.reprocessor.util.BlobUtil.getBlobCsv(BlobUtil.java:31) ~[classes/:na]
		at com.sams.cash.DLQProcessor.reprocessor.handler.PayoutMissedEventsHandler.process(PayoutMissedEventsHandler.java:31) ~[classes/:na]
		at com.sams.cash.DLQProcessor.reprocessor.client.QueueClientListener.onMessageAsync(QueueClientListener.java:68) ~[classes/:na]
		at com.microsoft.azure.servicebus.MessageAndSessionPump.lambda$receiveAndPumpMessage$0(MessageAndSessionPump.java:176) ~[azure-servicebus-3.6.6.jar:na]
		at java.base/java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1072) ~[na:na]
		at java.base/java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:478) ~[na:na]
		at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) ~[na:na]
		... 6 common frames omitted
```

**To Reproduce**

***Code Snippet***
```java
public String getBlobCsv(String fileName) {
    String directory = getDirectoryName(fileName);
    String csvString = "";
    BlobContainerClient blobContainerClient = blobClient.getContainerClient();
    PagedIterable<BlobItem> blobs;
    blobs = blobContainerClient.listBlobsByHierarchy(directory);
    log.info("Total Count of blobs  : {}",
        blobContainerClient.listBlobsByHierarchy(directory).stream().count());

    BlobItem blobItem = blobs.stream().filter(blob -> blob.getName().contains(fileName)).findFirst()
        .orElse(null);
    if (blobItem != null) {
      com.azure.storage.blob.BlobClient blobClient = blobContainerClient
          .getBlobClient(blobItem.getName());

      if (blobClient != null) {
        try (ByteArrayOutputStream outputStream = new ByteArrayOutputStream()) {
          blobClient.download(outputStream);
          csvString = new String(outputStream.toByteArray());
        } catch (IOException e) {
          log.info("Exception occurred while downloading blob : {}", e.getMessage());
        }
      }
    }
    return csvString;
  }
```

**Expected behavior**
I should be able to list blobs without any error , as before. There has been no functional change in code in past 1.5 years. Also 

**Screenshots**

**Setup (please complete the following information):**
 - IDE: [e.g. IntelliJ]
 - Library/Libraries: com.azure:azure-storage-blob:12.16.0 (groupId:artifactId:version)]
 - Java version: 11
 - App Server/Environment: 
 - Frameworks: Spring Boot

If you suspect a dependency version mismatch (e.g. you see `NoClassDefFoundError`, `NoSuchMethodError` or similar), please check out [Troubleshoot dependency version conflict article](https://aka.ms/azsdk/java/dependency/troubleshoot) first. If it doesn't provide solution for the problem, please provide:
- verbose dependency tree (`mvn dependency:tree -Dverbose`)
- exception message, full stack trace, and any available logs

**Additional context**

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [ ] Bug Description Added
- [ ] Repro Steps Added
- [ ] Setup information Added

**Comment by alzimmermsft** [comment]
Thank you for reporting this issue @shweta5595!

Looking at the stack trace you've included, I notice that `azure-core` `1.49.0` is being used alongside `azure-storage-blob` `12.16.0`.  This could be a possible issue with dependency skew, possibly resulting in this issue happening due to dependency conflicts. If possible, could you include the output of `mvn dependency:tree` on your application. This would help determine if the root issue is in dependency skew.

Additionally, if this is happening consistently, could you include the cause of the `HttpResponseException`? When this is thrown `HttpResponseBodyDecoder` should include the causal exception in `HttpResponseException` which should give a strong indication on what is failing during deserialization.

**Comment by shweta5595** [comment]
Hi , we can close this ticket , this was due to jackson version incompatibility .
